components/cam/src/physics/crm/RNG_MT.F90

```diff
@@ -0,0 +1,142 @@
+! Mersenne Twister Uniform Random Number Generator ((0,1) Real Number)
```

Comment:
There are some Random number generator algorithms under `cime/src/share/RandNum/`. I haven't carefully read this code but an RNG similar to this is at `cime/src/share/RandNum/src/mt19937/mersennetwister_mod.F90`. Unless there is a strong need to add a new one, I think we should use the existing RNGs.

Reply:
I initially planned to use an exiting RNG, but for this case I needed to get a single number at a time and ensure that the sequence was reproducible for each column. I remember concluding that the existing RNG routines were well equipped for this, so I just made a simpler implementation of the MT algorithm to handle this. 

In either case, this is only used for a special feature where the CRM orientation slowly rotates using a biased random walk in angular space. The idea was to get away from the fixed CRM orientation, which gives different systematic biases when comparing N/S to E/W. 

However, after testing this option I realized that the real problem is how the CRMs line up with the wind field, which causes cloud systems to "cycle" through the domain. There's also changes in model behavior when the CRM lines up with the shear. This is why the original Grabowski implementation constantly updated the CRM orientation based on the low-level wind. The N/S orientation minimizes the occurrence of having the CRM line up with the prevailing flow compared to the E/W orientation. The randomly changing orientation ends up representing the worst of both worlds by making the CRM line up with the prevailing flow everywhere at least once a day, which gives a mix of the worst systematic biases in both E/W and N/S options, rather than reducing them like I had hoped.

I'm still actively thinking about this problem, so I'd like to keep this code in place to revisit it, but it's not regularly used for the MMF. I'd like to revisit this in the next year or so and decide whether to keep it or not. If we decide to make this on by default then we can switch to using an existing RNG library.

Reply:
@singhbalwinder I agree, but I think we should defer these changes until after this PR is merged.

Reply:
Sounds good!